src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb

```diff
@@ -19,23 +19,64 @@ def initialize(azure_config, registry_endpoint, disk_manager, disk_manager2, azu
       @logger = Bosh::Clouds::Config.logger
     end
 
-    def create(bosh_vm_meta, location, vm_props, network_configurator, env)
+    def create(bosh_vm_meta, location, vm_props, disk_cids, network_configurator, env)
       # network_configurator contains service principal in azure_config so we must not log it.
-      @logger.info("create(#{bosh_vm_meta}, #{location}, #{vm_props}, ..., ...)")
+      @logger.info("create(#{bosh_vm_meta}, #{location}, #{vm_props}, #{disk_cids}, ..., ...)")
 
       instance_id = _build_instance_id(bosh_vm_meta, location, vm_props)
-
       resource_group_name = vm_props.resource_group_name
+      _check_resource_group(resource_group_name, location)
       vm_name = instance_id.vm_name
 
-      # When both availability_zone and availability_set are specified, raise an error
-      cloud_error("Only one of 'availability_zone' and 'availability_set' is allowed to be configured for the VM but you have configured both.") if !vm_props.availability_zone.nil? && !vm_props.availability_set.name.nil?
-      zone = vm_props.availability_zone
-      unless zone.nil?
-        cloud_error("'#{zone}' is not a valid zone. Available zones are: #{AVAILABILITY_ZONES}") unless AVAILABILITY_ZONES.include?(zone.to_s)
+      if vm_props.instance_type.nil?
+        calculated_instance_types = vm_props.instance_types.dup
+        @logger.debug("The cloud property 'instance_type' is nil. Will select one from '#{calculated_instance_types}', which are returned by calculate_vm_cloud_properties.")
+        is_persistent_disk_premium = false
+        disk_cids&.each do |disk_cid|
+          disk_id = DiskId.parse(disk_cid, @azure_config.resource_group_name)
+          if @use_managed_disks
```

Comment:
should not use `@use_managed_disks` here, we should not assume the disk type aligning with this attr.
This is quite complicated.  :(

Reply:
Let's make it pending.